src/daemon/bitcoind/interface.rs

```diff
@@ -1032,3 +955,88 @@ pub struct SyncInfo {
     pub ibd: bool,
     pub progress: f64,
 }
+
+#[derive(Clone, Debug)]
+pub struct ListUnspentEntry {
+    pub outpoint: OutPoint,
+    pub txo: TxOut,
+    pub label: Option<String>,
+    pub confirmations: u64,
+    pub derivation_index: Option<ChildNumber>,
+}
+
+impl From<&Json> for ListUnspentEntry {
+    fn from(utxo: &Json) -> Self {
+        let txid = utxo
+            .get("txid")
+            .map(|a| a.as_str())
+            .flatten()
+            .expect("API break, 'listunspent' entry didn't contain a string 'txid'.");
+        let txid = Txid::from_str(txid).expect("Converting txid from str in 'listunspent': {}.");
+        let vout = utxo
+            .get("vout")
+            .map(|a| a.as_u64())
+            .flatten()
+            .expect("API break, 'listunspent' entry didn't contain a valid 'vout'.");
+        let script_pubkey = utxo
+            .get("scriptPubKey")
+            .map(|s| s.as_str())
+            .flatten()
+            .map(|s| {
+                Script::from_str(s)
+                    .expect("API break, 'listunspent' entry didn't contain a valid script_pubkey")
+            })
+            .expect("API break, 'listunspent' entry didn't contain a string 'address'.");
+        let amount = utxo
+            .get("amount")
+            .map(|a| a.as_f64())
+            .flatten()
+            .expect("API break, 'listunspent' entry didn't contain a valid 'amount'.");
+        let value = Amount::from_btc(amount)
+            .expect("Could not convert 'listunspent' entry's 'amount' to an Amount")
+            .as_sat();
+        let confirmations = utxo
+            .get("confirmations")
+            .map(|a| a.as_u64())
+            .flatten()
+            .expect("API break, 'listunspent' entry didn't contain a valid 'confirmations'.");
+        let label = utxo
+            .get("label")
+            .map(|l| l.as_str())
+            .flatten()
+            .map(|l| l.to_string());
+        let mut derivation_index = None;
+        if let Some(d) = utxo.get("desc").map(|d| {
+            d.as_str()
+                .expect("API break, 'listunspent` entry contains a non-string desc")
+        }) {
+            // If we have a descriptor, we derive only once, so the derivation index must be
+            // between `/` and `]`
+            let derivation_index_start = d.find("/");
+            let derivation_index_end = d.find("]");
+            if let Some(s) = derivation_index_start {
+                if let Some(e) = derivation_index_end {
+                    // Also we always use normal derivation
+                    derivation_index = d[s + 1..e]
+                        .parse()
+                        .map(|d| ChildNumber::Normal { index: d })
+                        .ok();
+                }
+            }
+        }
+
+        ListUnspentEntry {
+            outpoint: OutPoint {
+                txid,
+                vout: vout as u32, // Bitcoin makes this safe
```

Comment:
I guess because of the maximum number of inputs by consensus? (Or maybe it's just a policy? Lol)
But I admit I copy-pasted the code from some other point in revaultd

Reply:
Yeah i noticed :)
Oh, i commented not on the right one. Yeah this one obviously because of the size limit.
My question was intended to be about another part where we assume block heights wont be >i32::MAX because so does bitcoind.